### PR TITLE
Adding dequantize_val and requantize_val

### PR DESCRIPTION
**Pull request by z-a-f**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#23909 Adding dequantize_val and requantize_val**

Differential Revision: [D16678276](https://our.internmc.facebook.com/intern/diff/D16678276)

```
$> ./build/bin/quantized_test
Running main() from ../third_party/googletest/googletest/src/gtest_main.cc
[==========] Running 4 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 4 tests from TestQTensor
[ RUN      ] TestQTensor.QuantDequantAPIs
[       OK ] TestQTensor.QuantDequantAPIs (0 ms)
[ RUN      ] TestQTensor.RoundingMode
[       OK ] TestQTensor.RoundingMode (0 ms)
[ RUN      ] TestQTensor.Item
[       OK ] TestQTensor.Item (0 ms)
[ RUN      ] TestQTensor.EmptyQuantized
[       OK ] TestQTensor.EmptyQuantized (0 ms)
[----------] 4 tests from TestQTensor (0 ms total)

[----------] Global test environment tear-down
[==========] 4 tests from 1 test case ran. (0 ms total)
[  PASSED  ] 4 tests.
```

### CODE REVIEW

**Comment by jerryzh168** [comment]
should we add some tests in https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/test/quantized_test.cpp

File: aten/src/ATen/test/quantized_test.cpp [hunk 1]
```diff
@@ -45,6 +46,24 @@ TEST(TestQTensor, QuantDequantAPIs) {
...
+              dequantize_val(qr.q_scale(), qr.q_zero_point(), qr_data[i]));
+  }
+
+  // Check for correct requantization
+  double new_scale = 2.0;
+  int64_t new_zero_point = 1;
+  Tensor reqr = at::quantize_linear(r, new_scale, new_zero_point, kQInt8);
+  auto reqr_data = reqr.data<qint8>();
+  for (auto i = 0; i < num_elements; ++i) {
+    reqr_data[i].val_ = requantize_val<quint8, qint8>(scale, zero_point,
```
Comment (jerryzh168) [inline aten/src/ATen/test/quantized_test.cpp]: can you fix the formatting?

File: aten/src/ATen/test/quantized_test.cpp [hunk 2]
```diff
@@ -82,7 +101,9 @@ TEST(TestQTensor, EmptyQuantized) {
   int zero_point = 10;
   int val = 100;
   int numel = 10;
-  Tensor q = at::_empty_affine_quantized({numel}, at::device(at::kCPU).dtype(kQUInt8), scale, zero_point);
+  Tensor q = at::_empty_affine_quantized({numel},
```
Comment (jerryzh168) [inline aten/src/ATen/test/quantized_test.cpp]: same here

File: aten/src/ATen/test/quantized_test.cpp (hunk 1 above)
Comment (zafartahirov) [reply aten/src/ATen/test/quantized_test.cpp]: Can you point to the actual formatting issue, please?

Comment (jerryzh168) [reply aten/src/ATen/test/quantized_test.cpp]: looks like the arguments are expanded into multiple lines and doesn't align. not sure whether this is just github display it this way or not. just make sure it looks ok in your editor

Comment (zafartahirov) [reply aten/src/ATen/test/quantized_test.cpp]: Resolved in person

**Review by jerryzh168 — APPROVED** [review]
Thanks!

**Pull request closed without merging**